tools/obscuroscan_v2/frontend/src/lib/batchList.js

```diff
@@ -0,0 +1,18 @@
+class BatchList {
+    constructor() {
+        this.items = [];
+    }
+
+    add(item) {
+        if (!this.items.includes(item)) {
+            this.items.push(item);
+        }
+    }
+
+    get() {
+        const startIdx = this.items.length - 5 > 0 ? this.items.length - 5 : 0;
+        return this.items.slice(startIdx);
```

Comment:
You can just return slice(-5), it supports negative indexes and behaves as you'd expect (the last 5 items or as many as there are).

Reply:
Good catch 👍